CIP-0118/README-implicit.md

```diff
@@ -0,0 +1,895 @@
+Category: Ledger
+Status: Proposed
+Authors:
+    - Polina Vinogradova <polina.vinogradova@iohk.io>
+    - Michael Peyton Jones <michael.peyton-jones@iohk.io>    
+    - Will Gould <will.gould@iohk.io>
+Implementors: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/780
+    - https://github.com/cardano-foundation/CIPs/pull/862
+Created: 2024-03-11
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We propose a set of changes that revolve around validation zones, a construct for allowing
+certain kinds of _underspecified transactions_. In particular, for the Babel-fees usecase
+we discuss here, we allow transactions that specify
+_part of a swap request_. A validation zone is a list of transactions such that
+earlier transactions in the list may be underspecified, while later transactions must
+complete all partial specifications. In the Babel-fees usecase, transactions do not 
+need to be balanced, i.e they do not require specifying where any missing funds come from, 
+or any extra funds go. A complete valid zone, however, must be balanced,
+with transactions providing all missing funds for other transactions in the zone, and 
+directing all "extra" funds to outputs/fees/etc.
+
+## Motivation: why is this CIP necessary?
+
+This CIP provides a partial solution to the problems described in CPS-15.
+In particular, it describes some ledger changes that allow intent settlement
+of intents that require “counterparty irrelevance”, including many of the swap use cases
+and dApp fee sponsorship. The motivation behind designing this solution is _Babel fees_,
+which is a requirement to support "paying transaction fees in a non-Ada currency".
+The Babel-fees usecase is a special case of a swap. Supporting swaps is very desirable functionality for cryptocurrency
+ledgers.
+
+Some of the primary contenders for implementing swaps
+are two-way off-chain communication (co-signing a single swapping transaction),
+smart contracts implementing DEX functionality, or ledger changes implementing
+an on-chain order matching DEX. We argue that our solution is the most versatile,
+least expensive for users, and maintains formal guarantees
+of the current Cardano ledger, among other benefits.
+
+We discuss additional possible solutions to problems outlined
+in CPS-15 that could make use of much of the same validation zones infrastructure,
+but with varying kinds of underspecified transactions.
+
+
+## Specification
+
+
+### Validation zones
+
+A validation zone is a list of transactions. This structure replaces singleton transactions
+as a unit of validation. Specifically, instead of singleton transactions,
+validation zones are
+
+- sent across the Cardano network
+- entered into a mempool, and
+- placed in block bodies.  
+
+When validating a zone (i.e. for placement in the mempool or when validating an
+incoming block), each transaction is checked against the current ledger
+state, however, additional checks
+are done at the zone level.
```

Comment:
Each transaction would **not** be checked against the current state. Instead, apply the first transaction to the current state (`s0`) to create `s1`. Apply the next transaction to `s1` to produce `s2`, and so on. The new states can be held in memory until they are squashed+merged or dropped.

Reply:
The wording is unclear here on my part - what you are describing is what i actually mean, i think. The state to which each transaction is applied is the state that was computed from the application of the previous one. The final state is just the result of the sequence of applications (intermediate states are not "remembered" by the ledger state in any way).